src/nile/signer.py

```diff
@@ -20,11 +20,14 @@ def __init__(self, private_key, network="testnet"):
         """Construct a Signer object. Takes a private key."""
         self.private_key = private_key
         self.public_key = private_to_stark_key(private_key)
-        self.chain_id = (
-            StarknetChainId.MAINNET.value
-            if network == "mainnet"
-            else StarknetChainId.TESTNET.value
-        )
+        if network == "mainnet":
+            self.chain_id = StarknetChainId.MAINNET.value
+        else:
+            self.chain_id = (
+                StarknetChainId.TESTNET2.value
+                if network == "goerli2"
+                else StarknetChainId.TESTNET.value
+            )
```

Comment:
```suggestion
        self.chain_id = {
            "mainnet": StarknetChainId.MAINNET.value,
            "goerli2": StarknetChainId.TESTNET2.value,
            "goerli": StarknetChainId.TESTNET.value
        }[network or "goerli"]
```

what do you think of something like this?

Reply:
That looks neat.

Reply:
We have the issue of not allowing custom network names though. But I think we can fix this when we add the nile.ini file with a network config. For example, I could want to configure a mainnet fork as "localhost-mainnet", and this won't work with this approach unless I add a try/catch block handling custom names.